File: PDRPC.Core/Utils/Logger.cs
using System;

internal static class Logger
{
    private static readonly string PREFIX = "[PDRPC]";

    public static void Info(object message)
    {
        Console.WriteLine($"{PREFIX}: {message}");
    }

    public static void Warning(object message)
    {
        Console.WriteLine($"{PREFIX}[WARN]: {message}");
    }

    public static void Error(object message)
    {
        Console.WriteLine($"{PREFIX}[ERROR]: {message}");
    }

    public static void Error(Exception exception)
    {
        Console.WriteLine($"{PREFIX}[ERROR]: {exception.Message}");
    }
}

File: PDRPC/src/logger.cpp
#include "logger.h"

void Logger::Info(std::string message)
{
	std::cout << "[PDRPC]: " << message << std::endl;
}

void Logger::Warning(std::string message)
{
	std::cout << "[PDRPC][WARN]: " << message << std::endl;
}

void Logger::Error(std::string message)
{
	std::cout << "[PDRPC][ERROR]: " << message << std::endl;
}
